R/execution.r

```diff
@@ -83,12 +62,13 @@ execute = function(request) {
     payload <<- lappend(payload, list(source='page', text=paste(text, collapse="\n")))
   })
 
-  send_plot = function(plotobj) {
-      tf = tempfile(fileext='.png')
-      do.call(png, c(list(filename=tf), get_plot_options()))
-      replayPlot(plotobj)
-      dev.off()
-      display_png(file=tf)
+  send_plot <- function(plotobj) {
+      params <- list()
+      for (mime in getOption('jupyter.plot_mimetypes')) {
+          r <- mime2repr[[mime]](plotobj)
+          params[[mime]] <- if (is.raw(r)) base64encode(r) else r
+      }
+      do.call(display_alternatives, params)
```

Comment:
Ok, I was going to argue that this should send a `execution_reply` but in python this also uses the `display` function :-/

But this then means that the kernel depends on the display package (due to the useage in sending plots) and the display package depends on the kernel (due to the actual sending of the `display_data` reply). I'm not sure if that's a problem in R.